bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/common/registry/AbstractRegistry.java

```diff
@@ -215,32 +219,34 @@ public E get(K key) {
         return null;
     }
 
+    public Entry<Provider<E>, E> getBoth(K key) {
```

Comment:
`getBoth` doesn't give good context for what it could be used. Besides, I am missing javadoc for this public method. Maybe `getValueWithProvider` would be a better name?